File: src/main/java/org/arbor/gtnn/data/materials/AdjustGTMaterials.java
package org.arbor.gtnn.data.materials;

import com.gregtechceu.gtceu.api.data.chemical.material.info.MaterialFlags;

import static com.gregtechceu.gtceu.common.data.GTMaterials.*;
import static org.arbor.gtnn.data.GTNNMaterials.*;

public class AdjustGTMaterials {
    public static void init() {
        NaquadahEnriched.addFlags(MaterialFlags.GENERATE_BOLT_SCREW);
        Brass.addFlags(MaterialFlags.GENERATE_DENSE);
        Aluminium.addFlags(MaterialFlags.GENERATE_DENSE);
        Steel.addFlags(MaterialFlags.GENERATE_DENSE);
        Lanthanum.addFlags(MaterialFlags.GENERATE_DENSE);
        Iridium.addFlags(MaterialFlags.GENERATE_DENSE);
        Lead.addFlags(MaterialFlags.GENERATE_DENSE);
        IronMagnetic.addFlags(MaterialFlags.GENERATE_PLATE);
        SteelMagnetic.addFlags(MaterialFlags.GENERATE_PLATE);
        NeodymiumMagnetic.addFlags(MaterialFlags.GENERATE_PLATE);
        SamariumMagnetic.addFlags(MaterialFlags.GENERATE_PLATE);
        NaquadahEnriched.addFlags(MaterialFlags.GENERATE_LONG_ROD);
        NickelZincFerrite.addFlags(MaterialFlags.GENERATE_LONG_ROD);
        BlueAlloy.addFlags(MaterialFlags.GENERATE_FRAME);
        Nichrome.addFlags(MaterialFlags.GENERATE_GEAR);
        Zeron100.addFlags(MaterialFlags.GENERATE_GEAR);
        Aluminium.addFlags(MaterialFlags.GENERATE_ROTOR);
        addGas(Oganesson);
        addGas(Calcium);
        addFluid(Californium);
        addFluid(Caesium);
        addDust(Praseodymium);
    }
}

File: src/main/kotlin/dev/arbor/gtnn/data/materials/AdjustGTMaterials.kt
package dev.arbor.gtnn.data.materials

import com.gregtechceu.gtceu.api.data.chemical.material.info.MaterialFlags.*
import com.gregtechceu.gtceu.api.data.chemical.material.properties.BlastProperty
import com.gregtechceu.gtceu.api.data.chemical.material.properties.OreProperty
import com.gregtechceu.gtceu.api.data.chemical.material.properties.PropertyKey
import com.gregtechceu.gtceu.api.data.tag.TagPrefix
import com.gregtechceu.gtceu.common.data.GTMaterials.*
import dev.arbor.gtnn.GTNN.getServerConfig
import dev.arbor.gtnn.common.item.properties.CatalystProperty
import dev.arbor.gtnn.data.GTNNMaterials.*
import dev.arbor.gtnn.data.GTNNPropertyKeys.CATALYST
import vazkii.botania.common.block.BotaniaBlocks
import vazkii.botania.common.item.BotaniaItems

object AdjustGTMaterials {
    fun init() {
        adjustOres()
        Neutronium.setProperty(
            PropertyKey.BLAST,
            BlastProperty(9000, BlastProperty.GasTier.HIGHEST, 491250, 144 * 20, -1, -1)
        )
        NaquadahEnriched.addFlags(GENERATE_BOLT_SCREW)
        Europium.addFlags(GENERATE_BOLT_SCREW)
        Brass.addFlags(GENERATE_DENSE)
        Aluminium.addFlags(GENERATE_DENSE)
        Steel.addFlags(GENERATE_DENSE)
        Lanthanum.addFlags(GENERATE_DENSE)
        Iridium.addFlags(GENERATE_DENSE)
        Lead.addFlags(GENERATE_DENSE)
        // IronMagnetic.addFlags(GENERATE_PLATE)
        SteelMagnetic.addFlags(GENERATE_PLATE)
        NeodymiumMagnetic.addFlags(GENERATE_PLATE)
        //SamariumMagnetic.addFlags(GENERATE_PLATE)
        NaquadahEnriched.addFlags(GENERATE_LONG_ROD)
        NickelZincFerrite.addFlags(GENERATE_LONG_ROD)
        BlueAlloy.addFlags(GENERATE_FRAME)
        Polybenzimidazole.addFlags(GENERATE_FRAME)
        Nichrome.addFlags(GENERATE_GEAR)
        Zeron100.addFlags(GENERATE_GEAR)
        Aluminium.addFlags(GENERATE_ROTOR)
        Iridium.addFlags(GENERATE_ROTOR)
        Iridium.addFlags(GENERATE_SMALL_GEAR)
        // EnrichedNaquadahTriniumEuropiumDuranide.setProperty(PropertyKey.WIRE, WireProperties(524288, 3, 2, false))
        addGas(Oganesson)
        addGas(Calcium)
        addFluid(Californium)
        addFluid(Caesium)
        addFluid(AmmoniumChloride)
        addDust(Praseodymium)

        OrangeMetal.setProperty(CATALYST, CatalystProperty(100))
    }

    private fun adjustOres() {
        if (getServerConfig().enableHarderPlatinumLine) {
            var oreProp: OreProperty = Nickel.getProperty(PropertyKey.ORE)
            oreProp.oreByProducts.clear()
            oreProp.setOreByProducts(Cobalt, Iron, PlatinumMetal)

            oreProp = Cooperite.getProperty(PropertyKey.ORE)
            oreProp.oreByProducts.clear()
            oreProp.setOreByProducts(Nickel, Nickel, Cobalt, PalladiumMetal)

            oreProp = Platinum.getProperty(PropertyKey.ORE)
            oreProp.oreByProducts.clear()
            oreProp.setOreByProducts(Nickel, Nickel, Cobalt, PalladiumMetal)
            oreProp.directSmeltResult = PlatinumMetal
        }
        if (getServerConfig().enableHarderNaquadahLine) {
            val oreProp = Naquadah.getProperty(PropertyKey.ORE)
            oreProp.oreByProducts.clear()
            oreProp.setOreByProducts(Sulfur, Barite, EnrichedNaquadahOxideMixture)
            oreProp.separatedInto.clear()
            oreProp.setSeparatedInto(EnrichedNaquadahOxideMixture)
        }
    }
}
